core/src/validator.rs

```diff
@@ -2315,8 +2322,19 @@ fn wait_for_supermajority(
     rpc_override_health_check: Arc<AtomicBool>,
     start_progress: &Arc<RwLock<ValidatorStartProgress>>,
 ) -> Result<bool, ValidatorError> {
-    match config.wait_for_supermajority {
-        None => Ok(false),
+    // wait_for_supermajority can be triggered either through commandline
+    // or when wen_restart is in WAIT_FOR_SUPERMAJORITY mode.
+    let (wait_for_supermajority, expected_bank_hash) = match config.wait_for_supermajority {
```

Comment:
If we're expecting the validator process to auto restart after phase 1,  seems reasonable to just have that top level logic pass in `wait_for_supermajority` instead of having to detect it here?

Reply:
Not sure what you mean, there is not other top level logic, it's just --wen_restart wen_restart_proto_path controlling the two phases. We don't want operators to wake up at night to change command line args.

Reply:
Thought about it more and I think your suggestion makes sense, changed.